queries.sql

```diff
@@ -22,4 +22,78 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+
+BEGIN;
+vet_clinic=*# UPDATE animals SET species = 'unspecified';
+
+ROLLBACK;
+
+-- nside a transaction:
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT; 
+
+
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+-- Inside a transaction:
+-- Delete all animals born after Jan 1st, 2022.
+-- Create a savepoint for the transaction.
+-- Update all animals' weight to be their weight multiplied by -1.
+-- Rollback to the savepoint
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+-- Commit transaction
+
+BEGIN;
+
+DELETE FROM animals
+WHERE Date_of_Birth > '2022-01-01';
+
+SAVEPOINT delete_age_greater;
+
+UPDATE animals
+SET weight_kg = weight_kg * (-1);
+
+ROLLBACK to SAVEPOINT delete_age_greater;
+
+UPDATE animals
+SET weight_kg = weight_kg * (-1)
+WHERE weight_kg < 0;
+
+
+-- Write queries to answer the following questions:
+
+-- How many animals are there?
+SELECT COUNT(*) AS total FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) AS total FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) as avg_weight FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, MAX(escape_attempts) as max_escape FROM animals GROUP BY neutered;
```

Comment:
- According to [this question](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Who%20escapes%20the%20most%2C%20neutered%20or%20not%20neutered%20animals%3F), I think it should be better to use `SUM(escape_attempts)` instead, to get the sum of all escape attempts from all animals types instead of having the escape attempts from only one animal. Kindly fix that too